diff --git a/src/ompl/datastructures/NearestNeighborsFLANN.h b/src/ompl/datastructures/NearestNeighborsFLANN.h
index 48c6921..d5cdfc5 100644
--- a/src/ompl/datastructures/NearestNeighborsFLANN.h
+++ b/src/ompl/datastructures/NearestNeighborsFLANN.h
@@ -42,8 +42,9 @@
 #  error FLANN is not available. Please use a different NearestNeighbors data structure.
 #else
 
-#include "ompl/datastructures/NearestNeighbors.h"
 #include "ompl/base/StateSpace.h"
+#include "ompl/datastructures/NearestNeighbors.h"
+#include "ompl/util/Exception.h"
 
 #include <flann/flann.hpp>
 
@@ -321,7 +322,7 @@ namespace ompl
     };
 
     template<>
-    void NearestNeighborsFLANN<double, flann::L2<double> >::createIndex(
+    inline void NearestNeighborsFLANN<double, flann::L2<double> >::createIndex(
         const flann::Matrix<double>& mat)
     {
         index_ = new flann::Index<flann::L2<double> >(mat, *params_);
